### PR TITLE
Add named export to type definitions

### PR DESCRIPTION
**Pull request by wight554**
* similar to https://github.com/fastify/fastify-cookie/pull/97

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -503,7 +503,21 @@ function getRedirectUrl (url) {
   }
 }
 
-module.exports = fp(fastifyStatic, {
+const fastifyStatic = fp(plugin, {
```
Comment (climba03003) [inline index.js:506]: You do not need to change this file. It is done inside `fastify-plugin`.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (wight554) [reply index.js:506]: Wdym? I just followed [this](https://github.com/fastify/fastify-cookie/pull/97) PR to add named export for consistency

Comment (climba03003) [reply index.js:506]: It is something already done by `fastify-plugin`. 
That means this package already have named export.

Comment (wight554) [reply index.js]: Oh well, so fastify-cookie has legacy implementation, it's clear for me now. Fixed

**Comment by wight554** [comment]
Main point of having this is that typescript looses typings when using default import of CJS module with tsconfig `module: "nodenext"` setting from TS 4.7:
```
import fastifyCookie
Argument of type 'typeof import("/.../node_modules/fastify-cookie/plugin")' is not assignable to parameter of type 'FastifyPluginCallback<any, Server> | FastifyPluginAsync<any, Server> | Promise<{ default: FastifyPluginCallback<any, Server>; }> | Promise<...>'.ts(2345)
```
More info here: https://github.com/microsoft/TypeScript/issues/48845

**Review by climba03003 — APPROVED** [review]
LGTM.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**